lib/chef/provider/systemd_unit.rb

```diff
@@ -58,20 +58,28 @@ def define_resource_requirements
 
       def systemd_unit_status
         @systemd_unit_status ||= begin
+          status = {}
+
           # Collect all the status information for a unit and return it at once
-          # This may fail if we are managing a template unit (e.g. with '@'), in which case
-          # we just ignore the error because unit status is irrelevant in that case
+          # This will fail if we are managing a template unit (e.g. with '@'), which is why
+          # we capture 'is-enabled' below
+          # This fails because '--system show' returns 'enabled' for *all* templated units
+          # if *any* of the templated units are enabled
+          # e.g. if daemon@env1 is enabled daemon@env2 will also show enabled,
+          # whether or not daemon@env2 is actually enabled
           s = shell_out(*systemctl_cmd, "show", "-p", "UnitFileState", "-p", "ActiveState", new_resource.unit_name, **systemctl_opts)
           # e.g. /bin/systemctl --system show -p UnitFileState -p ActiveState syslog.socket
           # Returns something like:
           # ActiveState=inactive
           # UnitFileState=static
-          status = {}
           s.stdout.each_line do |line|
             k, v = line.strip.split("=")
             status[k] = v
           end
 
+          status["is-enabled"] = shell_out(*systemctl_cmd, "is-enabled", new_resource.unit_name, **systemctl_opts).strip
```

Comment:
also some concerns here over perf and doing two shell outs again for everything.

Reply:
That's fair, but if incorrect results are returned, I'd argue we've overshot the target.

```bash
# /usr/bin/systemctl --system show -p UnitFileState -p ActiveState jboss@qa1 jboss@qa3
ActiveState=active
UnitFileState=enabled

ActiveState=active
UnitFileState=enabled

# systemctl disable jboss@qa3; systemctl daemon-reload
Removed symlink /etc/systemd/system/multi-user.target.wants/jboss@qa3.service.

# /usr/bin/systemctl --system show -p UnitFileState -p ActiveState jboss@qa1 jboss@qa3
ActiveState=active
UnitFileState=enabled

ActiveState=active
UnitFileState=enabled
```